db/compaction/compaction_picker_fifo.cc

```diff
@@ -177,6 +177,18 @@ Compaction* FIFOCompactionPicker::PickSizeCompaction(
               ,
               max_compact_bytes_per_del_file,
               mutable_cf_options.max_compaction_bytes, &comp_inputs)) {
+        if (FilesRangeOverlapWithCompaction(
+                {comp_inputs}, 0 /* output_level */,
+                Compaction::EvaluatePenultimateLevel(
+                    vstorage, ioptions_, comp_inputs.level /* start_level */,
+                    0 /* output_level */))) {
+          ROCKS_LOG_BUFFER(log_buffer,
+                           "[%s] FIFO compaction: Conflict with an ongoing "
+                           "compaction or file ingestion",
+                           cf_name.c_str());
+          return nullptr;
+        }
```

Comment:
[Note] Surprisingly, FIFO is not doing deletion compaction `* is deletion compaction */ false`, which means it will output files and can potentially conflict with file ingestion or refit level. Therefore I added the conflict check here. Same for below